### ISSUE TITLE
Run `azd pipeline config --provider azdo` pipeline missing Java SDK 

### ISSUE THREAD

**Issue by v-jiaodi** [Azure/azure-dev#3520]
**Describe the issue:**
Test `todo-java-mongo` template. Run `azd pipeline config --provider azdo`, get error as follow：
![image](https://github.com/Azure/azure-dev/assets/80496810/ab1fcccf-5d41-4a80-b5f3-4825c9ac707e)

**Repro Steps:**

1. Login in with command: `azd auth login`.
2. Execute command: `azd init -t todo-java-mongo -b staging`.
3. Run `azd pipeline config --provider azdo`.

**Environment:**
- Template: todo-java-mongo.
- OS: Windows, Mac, Codespace, Linux, DevContainer
- Azd version : 1.7.0-beta.1-daily.3582348  (commit https://github.com/Azure/azure-dev/commit/9a0cef9c79d83a5f3fe8cb8bc407a200139da62b)

**Expected behavior:**
Run `azd pipeline config --provider azdo`, pipeline log can pass.

@rajeshkamal5050  for notification.

**Comment by rajeshkamal5050** [comment]
@vhvb1989 can you triage this?

**Comment by vhvb1989** [comment]
the pipeline definition uses:

```yaml
  - task: JavaToolInstaller@0
    inputs:
      versionSpec: '17'
      jdkArchitectureOption: 'x64'
      jdkSourceOption: 'PreInstalled'
```

By using `jdkSourceOption: 'PreInstalled'`, it relies on JAVA SDK to be already pre-installed on the agent, which is the case for `Microsoft-hosted agents` (https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/java-tool-installer-v0?view=azure-pipelines#inputs)

It looks like the agent, in this case, doesn't have a pre-install JAVA sdk.

@v-jiaodi , can you show the output from JavaToolInstaller?:

![image](https://github.com/Azure/azure-dev/assets/24213737/e60f18fd-7081-4c47-a36e-71abc4213e00)

Note:
This is not azd or template's issue.  This looks like an issue with the Azure DevOps instace configuration.

**Comment by v-jiaodi** [comment]
@vhvb1989 Output from JavaToolInstaller:
![image](https://github.com/Azure/azure-dev/assets/80496810/c90ca201-5a3a-4acb-8c11-9dfb79c7b345)

**Comment by zhangjiale-64** [comment]
We also encountered the same problem in this manual test.

**Environment:**


- Template: todo-java-postgresql-terraform.
- OS: DevContainer
- Azd version: azd version 1.8.0-beta.1-daily.3649348 (commit https://github.com/Azure/azure-dev/commit/85f86be69c647f62cab6f5e42988d19df8bff03f)

**Comment by v-xuto** [comment]
@rajeshkamal5050 In the latest round of azd manual testing (1.9.2),  this issue still exists.

- Templates+OS: todo-java-postgresql-terraform (Devcontainer), ASA-Samples-Event-Driven-Application (Devcontainer), todo-java-mongo (Codespace)
- Branch: staging
- Azd version: azd version 1.10.0-beta.1-daily.3792048 (commit 13a7dd52710daa4ac5f0c17be6a77244005ac76e)

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd full regression test, this issue still exists.

Templates+OS: todo-java-mongo (Windows, Codespaces, Devcontainer, Linux, Mac.)
Branch: staging and pr/3976
Azd version: `azd version 1.10.0-beta.1-daily.3947259 (commit 5bf8568b1566723be8bc2259c578df4f931cf430)`

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Templates+OS: todo-java-mogno(Cloud Shell and Codespaces), ASA-Samples-Event-Driven-Application (Devcontainer).
- Branch: staging.
- Azd version: `azd version 1.10.0-beta.1-daily.4086794 (commit 9445b193c91acfa2d4bcae129cb7471ecc101e11)`.

**Comment by Menghua1** [comment]
@jongio  For the full regression test of [PR4258](https://github.com/Azure/azure-dev/pull/4258), this issue still exists.

- Templates:  todo-java-mongo 
- OS: Windows, Codespaces, Devcontainer, Linux, Mac
- Branch: staging
- Azd version: azd version 1.11.0-beta.1-pr.4140256 (commit b6402defbf4c4210855e14c114988010e0a51ed7)

**Comment by jongio** [comment]
Does this repro in main?

**Comment by v-xuto** [comment]
> Does this repro in main?

Yes, this issue can be reproduced when using `azd daily bits`.

- **Azd version**: azd version 1.11.0-beta.1-daily.4152592 (commit https://github.com/Azure/azure-dev/commit/e2ff8ba7dcc5b3f7e883213a2315a8b275294def)

**Comment by Menghua1** [comment]
@jongio Yes, this issue can also be reproduced in **main**. We tested the following four scenarios and they can all be reproduced:

- **Templates**:  todo-java-mongo 
- **OS**: Windows, Codespaces, Devcontainer, Linux, Mac
- **Test Scenario**: 
1. Templates under the `main` branch + azd version 1.11.0-beta.1-daily.4152592 (commit e2ff8ba7dcc5b3f7e883213a2315a8b275294def). 
2. Templates under the `main` branch + azd version 1.11.0-beta.1-pr.4140256 (commit b6402defbf4c4210855e14c114988010e0a51ed7). 
3. Templates under the `staging` branch + azd version 1.11.0-beta.1-daily.4152592 (commit e2ff8ba7dcc5b3f7e883213a2315a8b275294def). 
4. Templates under the `staging` branch + azd version 1.11.0-beta.1-pr.4140256 (commit b6402defbf4c4210855e14c114988010e0a51ed7).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Templates+OS: todo-java-mogno(Cloud Shell and Codespaces), ASA-Samples-Event-Driven-Application (Devcontainer).
- Branch: staging.
- Azd version: azd version 1.11.0-beta.1-daily.4152592 (commit e2ff8ba7dcc5b3f7e883213a2315a8b275294def).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Templates: todo-java-mogno.
- OS: `Cloud Shell`, `Codespaces`.
- Branch: Staging.
- Azd version: azd version 1.11.0-beta.1-daily.4247716 (commit https://github.com/Azure/azure-dev/commit/f4695c408eab9ea3cdedd1c3cb39aa01ad90123e).

**Comment by Menghua1** [comment]
@rajeshkamal5050 In the latest round of azd manual testing, this issue still exists.

- Templates: `todo-java-mogno`.
- OS: `Cloud Shell`, `Codespaces`.
- Branch: Staging.
- Azd version: azd version 1.12.0-beta.1-daily.4359062 (commit https://github.com/Azure/azure-dev/commit/6cd82ae9ff02e156a0056e994215191469442ab3).

**Comment by Menghua1** [comment]
It cannot be reproduced, I will close this issue.
